platform/x11/os_x11.cpp

```diff
@@ -469,6 +469,126 @@ void OS_X11::initialize(const VideoMode &p_desired, int p_video_driver, int p_au
 	power_manager = memnew(PowerX11);
 }
 
+XcursorImage* OS_X11::load_image (struct flist *list) {
+	XcursorImage *image;
+	png_structp png;
+	png_infop info;
+	png_bytepp rows;
+	FILE *fp;
+	int i;
+	png_uint_32 width, height;
+	int depth, color, interlace;
+	char *file;
+
+	png = png_create_read_struct (PNG_LIBPNG_VER_STRING, NULL, NULL, NULL);
```

Comment:
I dont really understand what are you trying to do here, but custom cursor functions should open an Image class and upload the data to the hardware, not open a PNG file. Godot can already load PNGs, so all that should be needed is parsing Image

Reply:
yeah, I tried before with Ref<Image> but I had some issues, I'll try again to see what issues I had and post here to see if you can help me.